modules/trainer/GenericTrainer.py

```diff
@@ -632,6 +641,68 @@ def sample_commands_fun():
 
                     self.one_step_trained = True
 
+                    # validation
+                    self.data_loader_validation.get_data_set().start_next_epoch()
+                    current_epoch_length_validation = self.data_loader_validation.get_data_set().approximate_length()
+                    if (self.config.validation
+                            and current_epoch_length_validation > 0
+                            and self.__needs_validate(train_progress)):
+
+                        torch_gc()
+
+                        step_tqdm_validation = tqdm(
+                            self.data_loader_validation.get_data_loader(),
+                            desc="validation_step",
+                            total=current_epoch_length_validation,
+                            initial=0)
+
+                        accumulated_loss_per_concept = {}
+                        concept_counts = {}
+                        mapping_seed_to_label = {}
+                        mapping_label_to_seed = {}
+
+                        for epoch_step_validation, batch in enumerate(step_tqdm_validation):
+
+                            if self.__needs_gc(train_progress):
+                                torch_gc()
+
+                            model_output_data = self.model_setup.predict(self.model, batch, self.config, train_progress)
```

Comment:
predict and calculate_loss should be executed in a `torch.no_grad()` context manager to speed up the calculation and preserve vram